types/handsontable/index.d.ts

```diff
@@ -401,8 +403,100 @@ declare namespace Handsontable {
+        /**
+         * Adds a listener (globally or locally) to a specified hook name.
+         *
+         * @param key Hook name.
+         * @param callback Callback function or an array of functions.
+         * @param context (optional) The context for the hook callback to be added - a Handsontable instance or leave empty.
+         */
+        add(key: string, callback: Function | Function[], context?: Handsontable): Object;
+
+        /**
+         * Returns a new object with empty handlers related to every registered hook name.
+         */
+        createEmptyBucket(): Object;
+
+        /**
+         * Deregisters a hook name (removes it from the list of known hook names).
+         *
+         * @param key Hook name.
+         */
+        deregister(key: string): void;
+
+        /**
+         * Destroy all listeners connected to the context. If no context is provided, the global listeners will be destroyed.
+         *
+         * @param context (optional) A Handsontable instance.
+         */
+        destroy(context?: Handsontable): void;
+
+        /**
+         * Get hook bucket based on the context of the object or if argument is undefined, get the global hook bucket.
+         *
+         * @param context (optional) A Handsontable instance.
+         */
+        getBucket(context?: Handsontable): Object;
+
+        /**
+         * Returns an array of registered hooks.
+         */
+        getRegistered(): string[];
+
+        /**
+         * Checks whether there are any registered listeners for the provided hook name.
+         *
+         * @param key Hook name.
+         * @param context (optional) A Handsontable instance.
+         */
+        has(key: string, context?: Handsontable): boolean;
+
+        /**
+         * Returns a boolean depending on if a hook by such name has been registered.
+         *
+         * @param key Hook name.
+         */
+        isRegistered(key: string): boolean;
+
+        /**
+         * Adds a listener to a specified hook. After the hook runs this listener will be automatically removed from the bucket.
+         *
+         * @param key Hook/Event name.
+         * @param callback Callback function.
+         * @param context (optional) A Handsontable instance.
+         */
+        once(key: string, callback: Function | Function[], context?: Handsontable): void;
+
+        /**
+         * Registers a hook name (adds it to the list of the known hook names). Used by plugins.
+         *
+         * @param key The hook name.
+         */
         register(key: string): void;
-        run(instace: Core, hookName: string, key?: any, value?: any): any;
+
+        /**
+         * Removes a listener from a hook with a given name.
+         *
+         * @param key Hook/Event name.
+         * @param callback Callback function.
+         * @param context (optional) A Handsontable instance.
+         */
+        remove(key: string, callback: Function, context?: Handsontable): boolean;
```

Comment:
Happy to remove the `Function` type. In terms of what to replace it with, the signatures of the `callback` vary dramatically (see [docs](https://docs.handsontable.com/0.31.2/Hooks.html#event:afterBeginEditing)) depending on which event you specify in the `key` string, so I for now, I think the best thing is to replace the type with:
```
(...params: any[]) => any
```

It would be a good later addition to the types library to also add types for all the callbacks, but that seems to me to be a large change that would make sense to do as a separate PR.

Also, it looks like the `Function` type is used in several places in the existing code (independent of this PR). I could prepare a separate PR to fix those as well if you'd like?